pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -69,21 +70,53 @@ class BaseMMM(BaseValidateMMM):
     _model_type: str = "BaseValidateMMM"
     version: str = "0.0.3"
 
+    @validate_call
     def __init__(
         self,
-        date_column: str,
-        channel_columns: list[str],
-        adstock_max_lag: int,
-        adstock: str | AdstockTransformation,
-        saturation: str | SaturationTransformation,
-        time_varying_intercept: bool = False,
-        time_varying_media: bool = False,
-        model_config: dict | None = None,
-        sampler_config: dict | None = None,
-        validate_data: bool = True,
-        control_columns: list[str] | None = None,
-        yearly_seasonality: int | None = None,
-        adstock_first: bool = True,
+        date_column: str = Field(..., description="Column name of the date variable."),
+        channel_columns: list[str] = Field(
+            min_length=1, description="Column names of the media channel variables."
+        ),
+        adstock_max_lag: int = Field(
+            ...,
+            gt=0,
+            description="Number of lags to consider in the adstock transformation.",
+        ),
+        adstock: str | InstanceOf[AdstockTransformation] = Field(
+            ..., description="Type of adstock transformation to apply."
+        ),
+        saturation: str | InstanceOf[SaturationTransformation] = Field(
+            ..., description="Type of saturation transformation to apply."
+        ),
+        time_varying_intercept: bool = Field(
+            False, description="Whether to consider time-varying intercept."
+        ),
+        time_varying_media: bool = Field(
+            False, description="Whether to consider time-varying media contributions."
+        ),
+        model_config: dict | None = Field(None, description="Model configuration."),
```

Comment:
It seems like the model_config will be either a Prior or the kwargs of HSGP tvps...
Let's make a Pydantic class for that will do some validations

Then we can have this more defined as 

`dict[str, Prior | HSGPKwargs]`

Hopefully that allows for dictionaries still that will be auto parsed

Reply:
Done! However, it complains when I use `dict[str, Prior | HSGPKwargs]`. I will investigate